### PR TITLE
SCMOD-8484: Add job failure roll up behaviour

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**
https://portal.digitalsafe.net/browse/SCMOD-8484

### CODE REVIEW

File: job-service-db/src/main/resources/procedures/changelog.xml [hunk 1]
```diff
@@ -199,6 +199,12 @@
                          schemaName="public">
         </createProcedure>
     </changeSet>
+    <changeSet id="create_process_failed_dependent_jobs_proc" runOnChange="true" author="Gamma Team">
+        <createProcedure path="procedures/processFailedDependentJobs.sql"
+                         procedureName="internal_process_failed_dependent_jobs"
+                         schemaName="public">
+        </createProcedure>
+    </changeSet>
```
Comment (dermot-hardy) [inline job-service-db/src/main/resources/procedures/changelog.xml:207]: @dgibson-microfocus I know you looked at the rules around this recently.  Remind me whether this is allowed or if it should be in a v3.3 folder.

File: worker-jobtracking/src/main/java/com/hpe/caf/worker/jobtracking/JobDatabaseProperties.java [hunk 2]
```diff
@@ -58,6 +58,11 @@ public static String getApplicationName() {
             : getPropertyOrEnvVar("JOB_DATABASE_APPNAME");
     }
     
+    public static boolean propagateFailures() {
```
Comment (dermot-hardy) [inline worker-jobtracking/src/main/java/com/hpe/caf/worker/jobtracking/JobDatabaseProperties.java]:
```suggestion
    public static boolean getPropagateFailures() {
```

```suggestion
    public static boolean getShouldPropagateFailures() {
```

File: job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    -- Ensure that no other `job` updates can run on these jobs until we've committed
+    -- Lock ALL possibly conflicting rows up-front to avoid deadlocks.
+    PERFORM NULL FROM job AS j
+    WHERE j.partition_id = in_partition_id
+        AND j.job_id IN (SELECT tmp_dependent_jobs.job_id FROM tmp_dependent_jobs)
+    ORDER BY j.partition_id, j.job_id
+    FOR UPDATE;
+
+    UPDATE job AS j
+    SET status = 'Failed', percentage_complete = 0.00
```
Comment (dermot-hardy) [inline job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql]: The failure details and last update date both need to be updated.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: job-service-db/src/main/resources/procedures/changelog.xml (hunk 1 above)
Comment (dgibson-microfocus) [reply job-service-db/src/main/resources/procedures/changelog.xml:207]: It should be in this changelog, not in it's own v3.3 folder. Having it in here means the latest version of the stored procedure will always be applied on upgrade and avoids any checksum and duplication problems if changed later on.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**